agent/containermetadata/parse_metadata.go

```diff
@@ -40,17 +43,32 @@ func (manager *metadataManager) parseMetadataAtContainerCreate(taskARN string, c
 // configuration and data then packages it for JSON Marshaling
 // Since we accept incomplete metadata fields, we should not return
 // errors here and handle them at this or the above stage.
-func (manager *metadataManager) parseMetadata(dockerContainer *docker.Container, taskARN string, containerName string) Metadata {
-	dockerMD := parseDockerContainerMetadata(taskARN, containerName, dockerContainer)
+func (manager *metadataManager) parseMetadata(dockerContainer *docker.Container, task *api.Task, containerName string) Metadata {
+	dockerMD := parseDockerContainerMetadata(task.Arn, containerName, dockerContainer)
 	return Metadata{
 		cluster:                 manager.cluster,
-		taskMetadata:            TaskMetadata{containerName: containerName, taskARN: taskARN},
+		taskMetadata:            TaskMetadata{containerName: containerName,
+			taskARN: task.Arn, taskDefinitionFamily: task.Family,
+			taskDefinitionRevision: parseTaskDefinitionRevision(task, containerName)},
 		dockerContainerMetadata: dockerMD,
 		containerInstanceARN:    manager.containerInstanceARN,
 		metadataStatus:          MetadataReady,
 	}
 }
 
+// parseTaskDefinitionRevision converts the task.Version from a string to an int.
+// this allows us to match the type used in our CLI:
+// https://docs.aws.amazon.com/cli/latest/reference/ecs/describe-task-definition.html
+// if anything bad happens, then 0 is returned (0 is not a valid task definition revision number),
+// so you must omitempty to omit this from the metadata JSON output.
+func parseTaskDefinitionRevision(task *api.Task, containerName string) int {
```

Comment:
Why do you need this method? Why not just use https://github.com/aws/amazon-ecs-agent/blob/edc3e26016e363d7e8e6c044b11bde9bbb8853a8/agent/api/task.go#L76?

Reply:
Because the type of the task.Version is a string, and the type in our CLI is an integer
(https://docs.aws.amazon.com/cli/latest/reference/ecs/describe-task-definition.html).
So to be consistent with the CLI, I make it a number here.

Reply:
Will make taskDefinitionRevision a string after further discussion.